backend/src/Backend/CacheServer.hs

```diff
@@ -153,8 +154,7 @@ cacheInfo storeDir = T.unlines
 
 
 storePathForHash :: Text -> Text -> IO (Maybe Text)
-storePathForHash storeDir hash = do
-  conn <- open nixSqliteDb
+storePathForHash storeDir hash = bracket (open nixSqliteDb) close $ \conn -> do
```

Comment:
`withConnection` (from `Database.SQLite.Simple`, which is also where `open` comes from) is defined as
```
\conn -> bracket (open connection) close
```

You should probably use that here instead of writing out the `bracket`